gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchTopHitsAnalyticsByEntrypointQueryAdapter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.repository.elasticsearch.v4.analytics.adapter;
+
+import io.gravitee.repository.log.v4.model.analytics.TopHitsAnalyticsByEntrypointQuery;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import java.util.HashMap;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class SearchTopHitsAnalyticsByEntrypointQueryAdapter {
+
+    public static final String ENTRYPOINT_ID_AGG = "entrypoint_id_agg";
+    public static final String FIELD = "field";
+    public static final String STATUS_RANGES = "status_ranges";
+
+    public static String adapt(TopHitsAnalyticsByEntrypointQuery query) {
+        var jsonContent = new HashMap<String, Object>();
+        var esQuery = buildElasticQuery(Optional.ofNullable(query).orElse(TopHitsAnalyticsByEntrypointQuery.builder().build()));
+        jsonContent.put("query", esQuery);
+        jsonContent.put("aggs", buildTopHistAnalyticsByEntrypointAggregation());
+        return new JsonObject(jsonContent).encode();
+    }
+
+    private static JsonObject buildTopHistAnalyticsByEntrypointAggregation() {
+        return JsonObject.of(
+            ENTRYPOINT_ID_AGG,
+            JsonObject.of(
+                // Group by entrypoint id (already filtered by <connector-type: entrypoint>
+                "terms",
+                JsonObject.of(FIELD, "entrypoint-id"),
```

Comment:
```suggestion
                JsonObject.of(FIELD, "entrypoint-id.keyword"),
```

We've just seen that we need to use this syntax to avoid "forcing a ES rollover index" et donc cette erreur 
https://gravitee.atlassian.net/browse/APIM-5102 (to be followed by the correction of this one )


src: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#before-enabling-fielddata